evalml/utils/nullable_type_utils.py

```diff
@@ -97,14 +97,49 @@ def _determine_downcast_type(col):
     Returns:
         LogicalType string to be used to downcast incompatible nullable logical types.
     """
-    downcast_matches = {
-        "BooleanNullable": ("Boolean", "Categorical"),
-        "IntegerNullable": ("Integer", "Double"),
-        "AgeNullable": ("Age", "AgeFractional"),
-    }
-
-    no_nans_ltype, has_nans_ltype = downcast_matches[str(col.ww.logical_type)]
+    no_nans_ltype, has_nans_ltype = DOWNCAST_TYPE_DICT[str(col.ww.logical_type)]
     if col.isnull().any():
         return has_nans_ltype
 
     return no_nans_ltype
+
+
+def _determine_fractional_type(logical_type):
+    """Determines what logical type to use for integer data that has fractional values imputed.
+    - IntegerNullable becomes Double.
+    - AgeNullable becomes AgeFractional.
```

Comment:
I think booleans should be on this list as well, right? 

But a follow up question about the boolean fractional value, is returning Categorical the desired behavior? How would we get to that scenario, and where is it used?

Reply:
nope - this is intended only for IntegerNullable and AgeNullable. We do have one place that I know of that turns booleans into fractional values in the time series imputer for target values, but that's not reachable from automl search, and I put up an issue to change that since it's nonsensical: https://github.com/alteryx/evalml/issues/4053

I was debating if I should make it explicit with an assertion that this shouldn't be used for BooleanNullable values, but it felt a little overkill for an internal util. I will, however, add a sentence to the docstring making it clear why BooleanNullable shouldnt be used.

Reply:
Okay, now seeing your other comment on just having the `equivalent` util return all types, I think that is a much better plan for this util as well. So I'll jsut update this one to return the same logical type if the type isnt int or age nullable

Reply:
made this change: 7a1b9f821